protocols/migration/migration_reports_4_0_0_to_reports_3_0_0.py

```diff
@@ -91,3 +94,194 @@ def migrate_genomic_feature(self, old_genomic_feature):
+        status_map = {
+            self.old_model.AdoptedStatus.notadopted: self.new_model.AdoptedStatus.not_adopted,
+            self.old_model.AdoptedStatus.adoptedin: self.new_model.AdoptedStatus.adoptedin,
+            self.old_model.AdoptedStatus.adoptedout: self.new_model.AdoptedStatus.adoptedout,
+        }
+        return status_map.get(old_status)
+
+    def migrate_person_karyotypic_sex(self, old_pks):
+        pks_map = {
+            self.old_model.PersonKaryotipicSex.UNKNOWN: self.new_model.PersonKaryotipicSex.unknown,
+            self.old_model.PersonKaryotipicSex.XX: self.new_model.PersonKaryotipicSex.XX,
+            self.old_model.PersonKaryotipicSex.XY: self.new_model.PersonKaryotipicSex.XY,
+            self.old_model.PersonKaryotipicSex.XO: self.new_model.PersonKaryotipicSex.XO,
+            self.old_model.PersonKaryotipicSex.XXY: self.new_model.PersonKaryotipicSex.XXY,
+            self.old_model.PersonKaryotipicSex.XXX: self.new_model.PersonKaryotipicSex.XXX,
+            self.old_model.PersonKaryotipicSex.XXYY: self.new_model.PersonKaryotipicSex.XXYY,
+            self.old_model.PersonKaryotipicSex.XXXY: self.new_model.PersonKaryotipicSex.XXXY,
+            self.old_model.PersonKaryotipicSex.XXXX: self.new_model.PersonKaryotipicSex.XXXX,
+            self.old_model.PersonKaryotipicSex.XYY: self.new_model.PersonKaryotipicSex.XYY,
+            self.old_model.PersonKaryotipicSex.OTHER: self.new_model.PersonKaryotipicSex.other,
+        }
+        return pks_map.get(old_pks)
+
+    def migrate_sex(self, old_sex):
+        sex_map = {
+            self.old_model.Sex.MALE: self.new_model.Sex.male,
+            self.old_model.Sex.FEMALE: self.new_model.Sex.female,
+            self.old_model.Sex.UNKNOWN: self.new_model.Sex.unknown,
+        }
+        return sex_map.get(old_sex, self.new_model.Sex.undetermined)
+
+    def migrate_analysis_panels(self, old_panels):
+        return None if old_panels is None else [self.migrate_analysis_panel(old_panel=old_panel) for old_panel in old_panels]
+
+    def migrate_analysis_panel(self, old_panel):
+        new_panel = self.convert_class(self.new_model.AnalysisPanel, old_panel)
+        new_panel.review_outcome = old_panel.reviewOutcome
+        new_panel.multiple_genetic_origins = old_panel.multipleGeneticOrigins
+        return self.validate_object(object_to_validate=new_panel, object_type=self.new_model.AnalysisPanel)
+
+    def migrate_files(self, old_files):
+        if old_files is None:
+            return None
+        if isinstance(old_files, list):
+            return [self.migrate_file(old_file=old_file) for old_file in old_files]
+        elif isinstance(old_files, dict):
+            return {key: self.migrate_file(old_file=old_file) for (key, old_file) in old_files.items()}
+
+    def migrate_file(self, old_file):
+        if old_file is None:
+            return None
+
+        if old_file is None:
+            return None
+        if isinstance(old_file.sampleId, list):
```

Comment:
we don't need this check in version 4 it is either a list or null.

Reply:
yeah good point, I was just copying what was going on in the 3 to 4 migration, this one still confuses me a bit...